h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -71,26 +71,28 @@ public boolean start(long timeout) {
             throw new IllegalStateException("Rabit tracker already started.");
         }
         RabitTrackerH2OThread trackerThread = new RabitTrackerH2OThread(this);
-        trackerThread.setDaemon(true);
-        trackerThread.start();
         this.trackerThread = trackerThread;
+        trackerThread.start();
         return true;
     }
 
     @Override
     public void stop() {
-        if(null != this.trackerThread) {
-            this.trackerThread.interrupt();
+        assert this.trackerThread != null;
+        try {
+                this.trackerThread.interrupt();
+            } catch (SecurityException e){
+                Log.err("Could not interrupt a thread in RabitTrackerH2O: " + trackerThread.toString());
+            }
             this.trackerThread.terminateSocketChannels();
             this.trackerThread = null;
 
-            try {
-                this.sock.close();
-            } catch (IOException e) {
-                throw new RuntimeException("Failed to close Rabit tracker socket.", e);
-            } finally {
-                this.port = 9091;
-            }
+
+        try {
+            this.sock.close();
+            this.port = 9091;
```

Comment:
Only reset the port to default when socket closing is successful. 

When unsuccessful, log an error and continue, as there may be enough resources to finish the task. The error will be in the logs still.